src/data/set/finite.lean

```diff
@@ -376,6 +390,11 @@ lemma finite_le_nat (n : ℕ) : finite {i | i ≤ n} := ⟨set.fintype_le_nat _
 
 lemma finite_lt_nat (n : ℕ) : finite {i | i < n} := ⟨set.fintype_lt_nat _⟩
 
+theorem exists_lt_modeq_of_infinite_nat {s : set ℕ} (hs : infinite s) {k : ℕ} (hk : 0 < k) :
+  ∃ (m ∈ s) (n ∈ s), m < n ∧ m % k = n % k :=
```

Comment:
`nat.modeq` is not available in this file. Do we have a better home for this lemma where both `set.finite` and `nat.modeq` are available?

Reply:
It could go in `pigeonhole.lean` (which does not import `modeq` currently, but this wouldn't be a problem to add it as it is not too low in the import hierarchy).